src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterUri.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class JsonValueConverterUri : JsonValueConverter<Uri>
+    {
+        public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out Uri value)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                value = default;
+                return false;
+            }
+
+            // TODO: use reader.GetUri() when https://github.com/dotnet/corefx/issues/38647 is implemented.
+            string uriString = reader.GetString();
+
+            try
+            {
+                value = new Uri(uriString);
```

Comment:
Should we support relative URI's?
> This constructor assumes that the string parameter references an absolute URI and is equivalent to calling the Uri constructor with UriKind set to Absolute. If the string parameter passed to the constructor is a relative URI, this constructor will throw a UriFormatException.

As it stands it can write relative URI's but not read them.

Reply:
This would be something the user would need to know statically \ up-front, correct? i.e. I don't think we want to try to dynamically figure out if a Uri is relative or not.

Assuming we don't add built-in support now, a custom converter could easily be authored to address relative URI's.

Built-in support would either create a second converter and make it public ("RelativeUriConverter") or make this public and add a bool property along with a custom attribute that can be applied to a property. In either case, this woud require an API review which is not feasible any longer for 3.0 due to schedule.

Reply:
Would we not just use (`Ury.TryCreate(uriString, UriKind.RelativeOrAbsolute, out Uri tmp)`)[https://docs.microsoft.com/en-us/dotnet/api/system.uri.trycreate?view=netframework-4.8#System_Uri_TryCreate_System_String_System_UriKind_System_Uri__] ?

Reply:
I was about to suggest exactly the same thing, then it would support either.

Reply:
Yes support both if it's that easy \ solution well-known.

I suppose, however, for some validation scenarios there are cases where you would want either a relative or absolute Uri and fail if not. That could be done through a custom converter.

Reply:
I think if someone did that it would be better to do that way, or custom code on the deserialized object, rather than throwing during the parsing which would be unrecoverable and hard to parse for validation reasons by application code.

Reply:
Thanks, I utilized the `Uri.TryCreate` method as suggested.